pkg/controller/node/node_controller.go

```diff
@@ -876,6 +888,30 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 		return err
 	}
 
+	observedGen := machineconfigpool.Generation
+	if ctrl.firstboot {
```

Comment:
How confident are we on this firstboot?

The daemon has issues where it fails on firstboot before completion (very rarely, e.g. lost leader election) and we can never complete items dependent on firstboot condition, so want to make extra sure of this.

Put another way, is this snippet crucial to MCO running? Or I guess it's just waiting for MCPs to get ready so transient errors don't pop up?

Reply:
I can try removing it and see if we fail. This was part of my other "hacks" we talked about. However, if the "MCP was modified plz try again" error is transient, this shouldn't be an issue.